src/Microsoft.Extensions.Hosting.Abstractions/IHostedService.cs

```diff
@@ -1,22 +1,20 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-namespace Microsoft.AspNetCore.Hosting
+namespace Microsoft.Extensions.Hosting
 {
     /// <summary>
     /// Defines methods for objects that are managed by the host.
     /// </summary>
     public interface IHostedService
     {
         /// <summary>
-        /// Triggered when the application host has fully started and the server is waiting
-        /// for requests.
+        /// Triggered when the application host has fully started.
         /// </summary>
         void Start();
```

Comment:
> Does this mean, there won't be a unified way to plug in logging, DI and configuration?

We're not merging the builders in 2.0. We have plans to take another look at this, but it's not going to land in time for 2.0. The goal of this change is:

- Make sure shared abstractions exist in a single place. Hosting.Extensions.Abstractions is the place where we add types that will be shared across the `WebHostBuilder` and `HostBuilder`.
- Move `IHostedService` to Hosting.Extensions.Abstractions. This is the new primitive which all things will build on top of. In an ideal world, http support is just an `IHostedService` and composes the http pipeline. Adding `IHostedService` in Hosting.Extensions.Abstractions also 
means other hosted services (like a background service `IHostedService`) can be shared in web and non-web scenarios.

The downside is that we need to duplicate a few extension methods (`ConfigureLogging`, `ConfigureServices`, `ConfigureAppConfiguration` etc) temporarily. The way I see it, if things work out, this host becomes the main host and we deprecate `WebHostBuilder`.

Here's the prototype I did a while back https://github.com/aspnet/Hosting/compare/davidfowl/uber-host.